src/freenet/support/SortedIntSet.java

```diff
@@ -93,44 +99,22 @@ public synchronized boolean contains(int num) {
 	 *            the item to be removed
 	 */
 	public synchronized boolean remove(int item) {
-		boolean ret = false;
 		int x = binarySearch(item);
 		if(x >= 0) {
-			if(x < length-1)
-				System.arraycopy(data, x+1, data, x, length-x-1);
-			data[--length] = Integer.MAX_VALUE;
-			ret = true;
+			return data.remove(x) != null;
```

Comment:
`ArrayList.remove(int index)` returns the value that was removed, or `null` if nothing was removed (note that the returned value is an `Integer` object or `null`). The original method is the equivalent of `ArrayList.remove(Object value)`, which returns whether the object was removed, as a boolean value.

That all said, considering removal of an element from the `ArrayList` is of linear complexity already… we might just use the latter method directly (with explicit cast to `Integer`).